Dockerfile

```diff
@@ -1,24 +1,34 @@
-FROM node:14.18.0-alpine as build
+FROM node:16.17.0-alpine as build
 
 ARG COMMIT_SHA=<not-specified>
-ENV NODE_ENV=production
+ENV NPM_CONFIG_CACHE="/tmp"
 
 WORKDIR /build-dir
 
 COPY package.json .
 COPY package-lock.json .
 
+# install also DEV dependencies to enable Typescript compilation
 RUN npm ci
 
 COPY . .
 
+# compile Typescript to Javascript
 RUN npm run build
 
+# remove files not needed in the final stage
+RUN rm -rf node_modules src tsconfig.tsbuildinfo tsconfig.json build.tsconfig.json
```

Comment:
removing node_modules could take some time on big projects; why don't we just copy what we need in the next stage?

Reply:
The idea is to not bring development packages (such as `typescript` and `tap`) into the final images. Maybe we should further evaluate the trade-off between creating a smaller image with less packages and the build time.

Reply:
Yeah I understand; what I suggesting is to prevent doing the rm (which may take some time on big projects) and only copy what is needed in the next layer (without copying the node_modules as you're doing here)

Reply:
Hi, sorry for my late answer. I made a change to copy only relevant files and removed the `rm` command in the `build` stage.